lua_cjson.c

```diff
@@ -738,7 +755,7 @@ static void json_append_data(lua_State *l, json_config_t *cfg,
     case LUA_TLIGHTUSERDATA:
         if (lua_touserdata(l, -1) == NULL) {
             strbuf_append_mem(json, "null", 4);
-        } else if (lua_touserdata(l, -1) == &json_empty_array) {
+        } else if (lua_touserdata(l, -1) == &json_array) {
```

Comment:
Since we changed the lightuserdata value in `cjson.empty_array` on L.1477 below, it is safe to switch this. I don't think keeping `cjson.empty_array` pointing to the `json_empty_array` pointer is especially better. I mainly updated it to point to the new `json_array` pointer because `cjson.empty_array_mt` and `json_empty_array` could disappear later on since they are currently being deprecated by this feature, which is likely a better approach.

Tests covering `cjson.empty_array` can be found in `tests/agentzh.t` as well, they have been left untouched compared to the master branch, ensuring backwards compatibility.

Reply:
@doujiang24 Of course, let me know if I missed anything that might make this change troublesome!

Reply:
@thibaultcha okay, got you, thanks for you explanation :)